spec/models/course/notification_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe Course::Notification, type: :model do
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    describe 'Create notifications' do
+      let(:course) { create(:course) }
+      let!(:course_users) { create_list(:course_user, 5, course: course) }
+      let(:user) { create(:user) }
+      let(:activity) do
+        create(:activity, actor: user, object: user, event: :tested)
+      end
+
+      context 'when create an email notification' do
+        let(:email_notification) do
+          build(:course_notification, course: course, activity: activity, notification_type: :email)
+        end
+
+        before do
+          allow_any_instance_of(ActivityMailer.view_context_class).
```

Comment:
I understand your idea. But there is a problem that I can not stub a method only for one instance of the email. Because this method is required when an email is creating. If I do not stub this method, it will raise an Missing Template error.